src/misc.jl

```diff
@@ -188,7 +188,8 @@ function inherit!(a::T, b::T) where T
 end
 
 
-isnothing(u) = u === nothing
+isnothing(::Nothing) = true
+isnothing(::Any) = false
```

Comment:
`isnothing` is in Compat now, so we shouldn't need this.  https://github.com/JuliaLang/Compat.jl/pull/654

Reply:
To make the code more readable without having to go find where `issomething` is defined, I've replaced all instances of `issomething` with `!isnothing` and removed both from misc.jl. I also updated Project.toml to specify 2.1 for `Compat` so that we know we have `isnothing`.